gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -242,134 +349,362 @@ static void
 {
-  /* GstTensorIf *self = GST_TENSOR_IF (object); */
-  /** @todo NYI! */
+  GstTensorIf *self = GST_TENSOR_IF (object);
+  g_critical ("Getting property for prop %d.\n", prop_id);
+
+  switch (prop_id) {
+    case PROP_CV:
+      g_value_set_enum (value, self->cv);
+      break;
+    case PROP_CV_OPTION:
+      break;
+    case PROP_OP:
+      break;
+    case PROP_SV:
+      break;
+    case PROP_SV_OPTION:
+      break;
+    case PROP_THEN:
+      break;
+    case PROP_THEN_OPTION:
+      break;
+    case PROP_ELSE:
+      break;
+    case PROP_ELSE_OPTION:
+      break;
+    case PROP_SILENT:
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
 }
 
 /**
- * @brief non-ip transform. required vmethod of GstBaseTransform.
+ * @brief Installs all the properties for tensor_if
+ * @param[in] gobject_class Glib object class whose properties will be set
  */
-static GstFlowReturn
-gst_tensor_if_transform (GstBaseTransform * trans,
-    GstBuffer * inbuf, GstBuffer * outbuf)
+static void
+gst_tensor_if_install_properties (GObjectClass * gobject_class)
 {
-  /* GstTensorIf *self = GST_TENSOR_IF (object); */
+  g_object_class_install_property (gobject_class, PROP_SILENT,
+      g_param_spec_boolean ("silent", "Silent", "Produce verbose output",
+          FALSE, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 
-  return GST_FLOW_ERROR; /** @todo NYI! */
+  g_object_class_install_property (gobject_class, PROP_CV,
+    g_param_spec_enum ("compared-value", "CV", "Compared value from input tensor(s)",
+        GST_TYPE_TENSOR_IF_CV, TIFCV_A_VALUE,
+        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_CV_OPTION,
+    g_param_spec_string ("compared-value-option", "CV_OPTION",
+        "Specify an element of the nth tensor or pick tensor ", "",
+        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_SV,
+    g_param_spec_string ("supplied-value", "SV",
+        " Supplied Value by user ", "",
+        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_OP,
+    g_param_spec_enum ("operator", "OP", "Comparison Operator",
+        GST_TYPE_TENSOR_IF_OP, TIFOP_EQ,
+        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_THEN,
+    g_param_spec_enum ("then", "THEN", "Action if it is TRUE",
+        GST_TYPE_TENSOR_IF_ACT, TIFB_PASSTHROUGH,
+        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_THEN_OPTION,
+    g_param_spec_string ("then-option", "THEN_OPTION",
+        "Pick tensor ", "", G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_ELSE,
+    g_param_spec_enum ("else", "ELSE", "Action if it is FALSE",
+        GST_TYPE_TENSOR_IF_ACT, TIFB_PASSTHROUGH,
```

Comment:
For else, "SKIP" might be appropriate for default action.

Reply:
Agree! Updated.